core/domain/search_services.py

```diff
@@ -244,7 +232,7 @@ def search_explorations(
               fetch, None otherwise. If an offset is returned, it will be a
               web-safe string that can be used in URLs.
     """
-    # The return type of 'platform_search_services.search()' method is
+    # Here we use cast because the return type of 'search()' method is
```

Comment:
As per the docstring, it can return both Dict and list.
https://github.com/oppia/oppia/blob/68bb129c7fe773b989670a59162534b8893789bb/core/platform/search/elastic_search_services.py#L185
Also, we cannot use overload here because we are returning Tuple here which is not mutable.

Reply:
Fixed by adding overloads.